Automattic/wp-calypso
# Plans 2023: Reindex files and package structure

Related to https://github.com/Automattic/wp-calypso/issues/78266

## Proposed Changes

* Reindexes and renames the files in `plans-features-2023-grid` to what the expected final package structure will be. 
* Outside of moving some logic into a main index file, there shouldn't be any other logical changes to the code.

`plans-features-2023-grid` has been renamed to `plans-grid` and the following new structure is introduced:
```
plans-grid/
├── components/
│   ├── comparison-grid/
│   │   ├── ...
│   │   └── index.tsx
│   ├── features-grid/
│   │   ├── ...
│   │   └── index.tsx
│   └── shared/
├── hooks/
├── lib/
├── ...
└── index.tsx
```

The main `index.ts` will next export the individual parts. Right now, it simply exports the current `plans-features-2023-grid` wrapper.

## Testing Instructions



This should hopefully be quick to test/verify and ship, as most things ought to be down to no build errors.

* Go to `/start/plans` and `/plans/[paid | free]` and confirm no breaking changes or console errors
* Comparison grid toggle works
* Features and comparison grid render fine. try resizing viewport
* changing monthly/yearly term works
* highlight labels, etc. render fine
* no styling changes from trunk

## Pre-merge Checklist
